sklearn/cluster/_hierarchical.pyx

```diff
@@ -449,3 +452,111 @@ def single_linkage_label(L):
         raise ValueError("Input MST array must be sorted by weight")
 
     return _single_linkage_label(L)
+  
+
+# Implements MST-LINKAGE-CORE from https://arxiv.org/abs/1109.2378
+@cython.boundscheck(False)
+@cython.nonecheck(False)
+cpdef np.ndarray[DTYPE_t, ndim=2] mst_linkage_core(
```

Comment:
This should just be a `def` since it's never called from cython

Reply:
Yes; done.